the-blue-alliance-ios/View Controllers/Match/MatchViewController.swift

```diff
@@ -39,18 +43,38 @@ class MatchViewController: MyTBAContainerViewController {
             tbaKit: tbaKit,
             userDefaults: userDefaults
         )
+        self.infoViewController.matchSummaryDelegate = self
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
+    
     // MARK: - View Lifecycle
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
         Analytics.logEvent("match", parameters: ["match": match.key!])
     }
+    
+}
 
+extension MatchViewController: MatchSummaryViewDelegate {
+    
+    func teamPressed(teamNumber: Int) {
+        guard let event = match.event else { return }
+        guard let redTeamKeys = match.redAlliance?.teams?.array as? [TeamKey] else { return }
```

Comment:
@ZachOrr Add a computed property on `Match` that aggregates all teams in a match. That way we can simplify this and do `match.teams`

Reply:
I also think we can tighten up this logic a little using Swift but I'm not entirely sure how yet... will look this evening when I'm at a compiler